plugin/pkg/scheduler/schedulercache/cache.go

```diff
@@ -106,6 +106,29 @@ func (cache *schedulerCache) List(selector labels.Selector) ([]*v1.Pod, error) {
 	return pods, nil
 }
 
+// PodFilter is a simple interface that helps filter out pods.
+// This is here only to avoid circular dependency between schedulercache and
+// algorithm. Ideally, it should have been in algorithm/types.go.
+type PodFilter interface {
```

Comment:
I'm not a fan of this approach. I would prefer defining:
```
type func(*v1.Pod) bool PodFilter
```

and passing explicitly nodeInfo.Filter() to FilteredList.

Or do you have any strong reason of not doing that?